### PR TITLE
Allow users to create notes for map layers

### PR DESCRIPTION
**Pull request by nyalldawson**
These notes are saved per layer, per project, and can be used as
place to store important messages for users of the project like
to do lists, etc.

Notes can be created via the "Add Layer Notes" action in the
layer right click menu. Any layers with notes will show a little
notepad indicator icon to alert users as to the notes. Clicking
the indicator will edit the note.

As discussed in https://github.com/qgis/QGIS-Enhancement-Proposals/issues/206

Funded by Alta Ehf

https://user-images.githubusercontent.com/1829991/115499492-e1235680-a2b2-11eb-97b9-dd20bf7ada99.mp4

### CODE REVIEW

**Comment by skampus** [comment]
will it possible to add other documents type as 'map layer note', like pdf?
thank you

stefano

**Comment by uclaros** [comment]
Nice!
Wouldn't it make sense to somehow inform the user that layer notes are stored in the project? (and not  in the layer/datasource)

**Comment by roya0045** [comment]
> Nice!
> Wouldn't it make sense to somehow inform the user that layer notes are stored in the project? (and not in the layer/datasource)

Isn't this the case for metadata in general?

**Comment by nyalldawson** [comment]
@skampus 
> will it possible to add other documents type as 'map layer note', like pdf?

That's out of scope for this work, but it IS possible to insert links into the text so you could potentially add a list of links to pdf documents stored on your computer/network/online/etc :+1:

**Comment by nyalldawson** [comment]
@uclaros 

> Wouldn't it make sense to somehow inform the user that layer notes are stored in the project?

Got any suggestions on how we could make the UI suggest this?

File: images/themes/default/mIndicatorNotes.svg [hunk 1]
```diff
@@ -0,0 +1 @@
....3398438 L 1.5 13.660156 C 1.5 14.125516 1.8744838 14.5 2.3398438 14.5 L 10.40625 14.5 L 14.5 10.40625 L 14.5 3.3398438 C 14.5 2.8744838 14.125516 2.5 13.660156 2.5 L 2.3398438 2.5 z" fill="none" stroke="#2c2c2c" stroke-opacity=".70588237"/><path d="M 5.5,11.5 H 8" fill="#2e4e72" fill-rule="evenodd" stroke="#2c2c2c" stroke-width=".999998" stroke-linecap="square" stroke-linejoin="bevel" stroke-opacity=".706"/><path d="M5.5 5.5h6M5.5 8.5h6M12.5 3.5v-2M9.5 3.5v-2M6.5 3.5v-2M3.5 3.5v-2" fill="#2e4e72" fill-rule="evenodd" stroke="#2c2c2c" stroke-linecap="square" stroke-linejoin="bevel" stroke-opacity=".706"/><path d="m 13.560003,2.935202 -11.0570028,0.055 c -0.3260001,-0.035 -0.5630002,0.089 -0.5020001,0.5 v 10.015625 c -0.064,0.594 0.3963172,0.508116 0.5,0.5 H 3.0000003 V 3.990202 H 14.000001 v -0.5 c 0,-0.178 0.04698,-0.55 -0.499998,-0.5 z" fill="gray" fill-opacity="0.424"/><path d="M 9.9999998,10.499999 H 14" fill="#2e4e72" fill-rule="evenodd" stroke="#2c2c2c" stroke-width=".999997" stroke-linejoin="bevel" stroke-opacity=".706"/><path d="m 10.500002,11 v 3" fill="#2e4e72" fill-rule="evenodd" stroke="#2c2c2c" stroke-width=".999996" stroke-linejoin="bevel" stroke-opacity=".706"/></svg>
```
Comment (nirvn) [inline images/themes/default/mIndicatorNotes.svg:1]: Look at this cute little notepad outline!

**Review by nirvn — APPROVED** [review]

**Comment by Gustry** [comment]
Thanks @nyalldawson and Alta Ehf for this feature.

> Sponsored by Alta Ehf

Arf, the changelog parser will parse "Funded by", not "Sponsored by" :)
I will go and update in the code when I will have some quick spare time. But if in the meantime, you can switch to "Funded by" just in case ;-)

**Comment by nyalldawson** [comment]
Thanks for the heads up @Gustry !

**Comment by Gustry** [comment]
PR done : https://github.com/kartoza/prj.app/pull/1333
Not sure when it will be merged and deployed on the server.

Sponsored by [name] [url]
Funded by [name] [url]

will work.

**Comment by nyalldawson** [comment]
Great! Thanks @Gustry !

**Pull request merged**